### PR TITLE
36 add serotypefinder results parser

### PR DESCRIPTION
**Pull request by ryanjameskennedy**
prp has been tested on ecoli samples

### CODE REVIEW

File: prp/models/phenotype.py [hunk 1]
```diff
@@ -50,9 +52,10 @@ class ElementStressSubtype(Enum):
 
 
 class ElementAmrSubtype(Enum):
-    """Categories of resistance and virulence genes."""
+    """Categories of resistance genes."""
 
     AMR = "AMR"
+    POINT = "POINT"
```
Comment (mhkc) [inline prp/models/phenotype.py:58]: Is this a SNV?

File: prp/models/phenotype.py [hunk 2]
```diff
@@ -209,5 +224,5 @@ class ElementTypeResult(BaseModel):
     """
 
     phenotypes: Dict[str, List[str]]
-    genes: List[Union[AmrFinderResistanceGene, AmrFinderGene, ResfinderGene, VirulenceGene]]
+    genes: List[Union[AmrFinderResistanceGene, AmrFinderGene, ResfinderGene, VirulenceGene, SerotypeGene]]
```
Comment (mhkc) [inline prp/models/phenotype.py:227]: Should this be described as a phenotypic result isnt this for typing?

**Review by mhkc — CHANGES_REQUESTED** [review]
Code wise it looks good but I had some questions on how the information should be formatted.

File: prp/models/phenotype.py (hunk 1 above)
Comment (ryanjameskennedy) [reply prp/models/phenotype.py:58]: This is the way it's reported in amrfinderplus. We're using their nomenclature right?

File: prp/models/phenotype.py (hunk 2 above)
Comment (ryanjameskennedy) [reply prp/models/phenotype.py:227]: Yeah but typing is still technically a phenotype. VirulenceGene is used for `stx` typing but it's in the `typing.py` model. I just copied that logic...

File: prp/models/phenotype.py [hunk 3]
```diff
@@ -141,19 +150,27 @@ class ResistanceGene(GeneBase):
     phenotypes: List[PhenotypeInfo] = []
 
 
+class SerotypeGene(GeneBase):
+    """Container for serotype gene information"""
+
+    depth: Optional[float] = Field(
```
Comment (ryanjameskennedy) [inline prp/models/phenotype.py:156]: Should I remove this `depth`variable as serotypefinder doesn't produce this in the output (same with virulencefinder yet it's included in `VirulenceGene`)? @mhkc

File: prp/models/phenotype.py (hunk 2 above)
Comment (mhkc) [reply prp/models/phenotype.py]: Yes but its used for tryping. Depending if we categorize a method as a phenotype prediction or typing method determines where and how its going to be displayed in bonsai.

**Review by mhkc — APPROVED** [review]

File: prp/models/phenotype.py (hunk 1 above)
Comment (ryanjameskennedy) [reply prp/models/phenotype.py:58]: It's still element_type="AMR" & element_subtype = "POINT"

**Pull request merged**